### PR TITLE
Run self tests as nobody for at least one CI target

### PR DESCRIPTION
**Pull request by superm1**
Beyond the failures that happened in ATA and NVME plugins, other parts in the codebase are making writes into `data/tests` or `plugins/synaptics-mst/tests`.  We shouldn't be allowing this at all.  Fix those and then enforce that everything touched is in `build/` or `/tmp` during the run.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/synaptics-mst/fu-synaptics-mst-device.c
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst-device.c:85]: Is this safe? Do we ever set the udev dev after the mst device has been set up?

Comment (superm1) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:85]: We shouldn't be at least..  Synaptics MST device inherits from a udev device, and that's the thing we're setting flags on

**Review by hughsie — APPROVED** [review]
Lets wait for CI, but LGTM, thanks for figuring this one out.

**Pull request merged**
